### ISSUE TITLE
Enable privateFor for Public Contracts

### ISSUE THREAD

**Issue by web3rover** [Consensys-Incorporated/quorum#138]
Is there a way I can change the source code to enable private transactions for public contracts. I am aware of the outcome. But I need this feature.

**Comment by web3rover** [comment]
@bas-vk @patrickmn @obscuren can you help on this?

**Comment by patrickmn** [comment]
It's possible, but would require substantial changes.

Can you describe the problem you're trying to solve in more detail either here or to quorum_info@jpmorgan.com? Perhaps we can suggest an alternative that does not require such modifications.

**Comment by web3rover** [comment]
@patrickmn If I deploy a contract as private contract then I am unable to add new members to the contract. It will be only visible to them who are in the privateFor array. So I want to make a contract public and make private transactions to it.

**Comment by patrickmn** [comment]
The ability to track contract participants as well as add new ones when desired is something we're currently working on:

  - Easier way to keep track of privateFor arrays for a contract https://github.com/jpmorganchase/quorum/issues/110
  - Ability to reveal a contract to/add a participant to a contract post-creation https://github.com/jpmorganchase/quorum/issues/142

Once this feature lands, that would be the best way to address your problem, as far as I can tell.

May I suggest creating a new private contract with the new set of participants for now? Old contracts could have a field indicating the address of the contract that replaces them.

**Comment by web3rover** [comment]
@patrickmn Thanks for the suggestion. Looking forward for the upcoming features. 👍

**Comment by jiten-balla** [comment]
@patrickmn May I know the dev status for this issue. Any version is rolled out?

**Comment by fixanoid** [comment]
@jiten-balla this issue is currently closed without an implementation we intend to provide since the functionality is a bit ambiguous. Please open a new ticket if you need more details.
